### PR TITLE
Actually fail if staleness fail

### PR DESCRIPTION
**Pull request by mkruskal-google**
PiperOrigin-RevId: 610833812

### CODE REVIEW

**Comment by mkruskal-google** [comment]
Skipping presubmits, since this workflow doesn't even run

File: .github/workflows/staleness_check.yml [hunk 1]
```diff
@@ -56,8 +56,7 @@ jobs:
           bash: >
             set -ex;
             if [[ -z $COMMIT_TRIGGERED_RUN || -z $MAIN_RUN ]]; then
-            bazel query 'attr(tags, "staleness_test", //...)' | xargs bazel test $BAZEL_FLAGS ||
-            echo "Please run ./regenerate_stale_files.sh to regenerate stale files";
+            bazel query 'attr(tags, "staleness_test", //...)' | xargs bazel test $BAZEL_FLAGS;
```
Comment (haberman) [inline .github/workflows/staleness_check.yml:59]: The message seems useful. I didn't know the right command to run to fix this.

Would this work?

```
bazel query 'attr(tags, "staleness_test", //...)' | xargs bazel test $BAZEL_FLAGS || {
  echo "Please run ./regenerate_stale_files.sh to regenerate stale files";
  exit 1;
}
```

Comment (mkruskal-google) [reply .github/workflows/staleness_check.yml:59]: I think so, but these tests are so fragile I wanted to keep it as simple as possible :(

**Review by haberman — APPROVED** [review]

File: .github/workflows/staleness_check.yml (hunk 1 above)
Comment (mkruskal-google) [reply .github/workflows/staleness_check.yml:59]: Yea that sgtm

**Pull request merged**